From 1b126d99b714e3c929616c8457c26a4dd68e6e45 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sat, 31 Dec 2022 14:19:43 +0100
Subject: [PATCH] apk_defines: use unsigned operand in BIT

This fixes undefined behavior at least in database, where BIT is
used with 31 (as APK_MAX_REPOS is 32) which is not representable
with a signed integer.
---
 src/apk_defines.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/apk_defines.h b/src/apk_defines.h
index 66661be..604ea44 100644
--- a/src/apk_defines.h
+++ b/src/apk_defines.h
@@ -17,7 +17,7 @@
 #include <time.h>
 
 #define ARRAY_SIZE(x)	(sizeof(x) / sizeof((x)[0]))
-#define BIT(x)		(1 << (x))
+#define BIT(x)		(1U << (x))
 #define min(a, b)	((a) < (b) ? (a) : (b))
 #define max(a, b)	((a) > (b) ? (a) : (b))
 
-- 
2.38.1

